src/main/java/org/b612foundation/adam/datamodel/BatchPropagation.java

```diff
@@ -0,0 +1,88 @@
+package org.b612foundation.adam.datamodel;
+
+import java.util.List;
+import java.util.Objects;
+
+public class BatchPropagation extends AdamObject {
+
+  /** Parameters used to generate the ephemerides of propagations in this batch. */
+  private PropagationParameters templatePropagationParameters;
+
+  /** Human-readable description of this object. */
+  private String description;
+
+  /** Type of this batch propagation. Options: HYPERCUBE, SINGLE. Type may indicate presence of custom fields in OPM. */
+  private String type;
+
+  /** Summary of the results. */
+  private String summary;
+
+  /** Uuids of single propagations carried out as part of this batch. */
+  private List<String> propagationUuids;
```

Comment:
I am not convinced this should be here at all. Depending on the wiring, there may not even been child uuids.

Reply:
Let's talk about this one some more. I put it here because I imagine that our soon-implementation won't support more than keep all ephemerides or keep no ephemerides, and this seemed like the most straightforward way to expose the actual ephemeris files. I agree that in the long term, this will probably go away in favor of something that allows us to reproduce runs on-demand, such as a list of opaque child parameters. But I don't think it'll be hard to change then, so I chose what seemed to me like the simplest option for now.

Reply:
Actually a very concrete reason not to do this is that I'm not sure you can have a list field in mysql, and I can't see any way to have a field that is a list of strings that are foreign keys. So the straightforward thing might be to keep what we have now (single prop references batch prop, probably optionally), and probably a call to get all single propagations associated with the batch minus ephemerides, then allow fetching ephemeris for a single propagation.

Reply:
Yeah, having some way of finding a child's parent is probably better. The only interesting case is if we support child reuse, but that's (1) kinda weird to begin with, and (2) can be done via a separate 4-column table (parent type-uuid, child type-uuid).

In any case, astrodynamics library doesn't need to know about that kind of stuff anyway. It doesn't need to know about uuids in general, but we agreed that having that one field here is an acceptable distraction that makes other things easier.

Reply:
True. Yeah, we can figure out what we want for child-parent mapping. We'll definitely need something for the pubsub framework, but that might be in the form of continuations or something like that. Question is whether that works well enough for things like AdamFront retrieving all propagations in a batch, or whether we want to allow another expression of the linkages.